instructions/docs/add-the-encryption-sdk.md

```diff
@@ -0,0 +1,230 @@
+* `search`: This operation accepts a metadata key to search for. It then queries DynamoDB for the `set` of documents in the Document Bucket that have context matching that key. This operation then returns that set of document identifiers and their metadata.
+    * Once the desired document or documents have been identified with the returned metadata, the document identifiers can be passed to `retrieve` to actually fetch the documents.
+
+This is a start for sharing, storing, and searching documents of a variety of types. But what about sensitive documents? Or protecting, say, important configuration files from accidental corruption during storing or retrieving?
+
+Now you will add the AWS Encryption SDK to take an extra step of protection: client-side encrypting the Document Bucket document before it is transmitted off of the host machine. You will use KMS to provide a `data key` for each document, using a CMK that you set up in [Getting Started](./getting-started.md) when you deployed your stacks.
+
+## Make the Change
+
+### Starting Directory
+
+Make sure you are in the `exercises` directory for the language of your choice:
+
+```Java
+~/environment/workshop/exercises/java
+```
+
+```NodeJS
+~/environment/workshop/exercises/node-javascript
+```
+
+```Python
+~/environment/workshop/exercises/python
+```
+
+`cd` into the `add-esdk-start` directory.
+
+### Step 1: Add the ESDK Dependency
+
+Look for `ADD-ESDK-START` comments to help orient yourself in the code.
+
+Start by adding the Encryption SDK dependency to the code.
+
+```NodeJS
+TODO
+```
+
+```Python
+# Edit src/document_bucket/__init__.py
+
+import aws_encryption_sdk
+
+# Save and exit
+# Edit src/document_bucket/api.py
+
+# ADD-ESDK-START
+import aws_encryption_sdk
+from aws_encryption_sdk import KMSMasterKeyProvider
+
+# Add a Master Key Provider to your __init__
+# ADD-ESDK-START
+def __init__(self, bucket, table, mkp):
+    self.bucket = bucket
+    self.table = table
+    # ADD-ESDK-START
+    self.master_key_provider
+
+# Save and exit
+```
+
+#### What Just Happened
+
+You just imported a dependency on the AWS Encryption SDK library in your code.
+
+You also changed the API to expect that a Keyring or Master Key Provider will be passed to your code to use in `store` and `retrieve` operations.
+
+### Step 2: Add Encryption to `store`
+
+Now that you have the AWS Encryption SDK imported, start encrypting your data before storing it.
+
+```NodeJS
+TODO
+```
+
+```Python
+# Edit src/document_bucket/api.py
+def store(self, data: bytes, context: Dict[str, str] = {}) -> PointerItem:
+    # ADD-ESDK-START
+    encrypted_data, header = aws_encryption_sdk.encrypt(
+        source=data,
+        key_provider=self.master_key_provider,
+    )
+```
+
+#### What Just Happened
+
+You just started encrypting data client-side with the AWS Encryption SDK and KMS!
+
+Now, before storing your data in the Document Bucket, you use the AWS Encryption SDK to:
```

Comment:
I'd suggest dropping the "your", but maybe purely stylistic

Reply:
I'd also suggest saying "the application" instead of "you", but again… perhaps stylistic

Reply:
Fixed